### ISSUE TITLE
EnderPearl cooldown doesn't occur in offhand.

### ISSUE THREAD

**Issue by DahFrozen** [Sudzzy/FBasics#43]
Hi, 
in 1.9+ players can bypass the Ender Pearl cooldown by throwing it from their offhand. I would appreciate it a lot if you could patch this :)

Thanks,
Frozon.

**Comment by RoboMWM** [comment]
Is your server running 1.9 or later? If so, sounds like the cooldown they added is client-side then

**Comment by DahFrozen** [comment]
The server is running 1.10. 

The vanilla client-side cooldown takes place on both the off-hand and main-hand - it has nothing todo with this issue. The issue is that the plugin doesn't listen for when a player throws an EnderPearl from their offhand.

**Comment by DahFrozen** [comment]
Bump.

**Comment by RoboMWM** [comment]
Bumping is 100% useless for any issue tracker. This isn't a forum.

**Comment by sgdc3** [comment]
@RoboMWM stop trying to be the issue tracker master, if you really want to help there go to check the code and maybe PR an hotfix. I would do that by myself but i'm from mobile.

**Comment by sgdc3** [comment]
Look! In found the issue in 3 seconds: https://github.com/Sudzzy/FBasics/blob/master/FBasics/src/main/java/org/originmc/fbasics/listener/EnderpearlListener.java#L128

**Comment by RoboMWM** [comment]
> stop trying to be the issue tracker master, 

Says the guy scolding me about how to deal with issues on an issue tracker

**Comment by DahFrozen** [comment]
@sgdc3 So will we need to get @Sudzzy to fix it or can you edit the code yourself? I wish I could help, but I don't have any experience in Java :/

**Comment by RoboMWM** [comment]
> Look! In found the issue in 3 seconds: https://github.com/Sudzzy/FBasics/blob/master/FBasics/src/main/java/org/originmc/fbasics/listener/EnderpearlListener.java#L128

Just had time to look at the code now (yes, I'm busy Tue-Thurs) and sgdc3 is wrong.

It seems Sudzzy already added 1.9 offhand support as per this commit: https://github.com/Sudzzy/FBasics/commit/1413c77f63ca45beccefc4d83b8a427cc3571c3e

So add a specific, reproducible test case and/or perhaps a video, of what is happening vs. what should happen. Also include `/version Fbasics` and your config.

**Comment by sgdc3** [comment]
@RoboMWM Ok, you're right about the right click thing, i was from mobile and i completely forgot how enderpearl in offhand can be throwed. I was only trying to say that github isn't a proper issue tracker, it is more similar to a forum, and bumping may be useful to keep an issue into someone's notifications (like your and mine)

**Comment by DahFrozen** [comment]
@RoboMWM 

>version
[17:00:53 INFO]: This server is running Paper version git-Paper-863 (MC: 1.10.2) (Implementing API version 1.10.2-R0.1-SNAPSHOT)
[17:00:53 INFO]: You are 1 version(s) behind

>fbasics version
[17:01:03 INFO]: FBASICS // Using version 2.1.8
[17:01:03 INFO]: FBASICS // Written by Sudzzy
[17:01:03 INFO]: FBASICS // Website: http://originmc.org/

Players can bypass the Enderpearl cooldown when they put their Enderpearls in their off-hand and right click. I'm using the latest FBasics.jar from the Spigot page.

**Comment by RoboMWM** [comment]
> Players can bypass the Enderpearl cooldown when they put their Enderpearls in their off-hand and right click. 

Still missing config.

**Comment by DahFrozen** [comment]
@RoboMWM Here is my config. 
http://pastebin.com/fiQSeUyT

**Comment by DahFrozen** [comment]
@RoboMWM Any news? Are you able to reproduce the issue?

**Comment by DahFrozen** [comment]
Bump - this is actually quite important to the gameplay of my server.

**Comment by RoboMWM** [comment]
At the moment I don't use Fbasics, and when I did, I only used a subset of Enderpearl cooldown features. I haven't had time to look into it, especially considering that it looks like this _should've_ been implemented.

**Comment by DahFrozen** [comment]
@RoboMWM ah okay. I would appreciate it a lot if you got back to me relating if the cooldown in the offhand worked or not.

**Comment by DahFrozen** [comment]
Bump

**Comment by DahFrozen** [comment]
Bump

**Comment by RoboMWM** [comment]
FYI, I've unsubscribed to this issue as I pretty much don't use enderpearl cooldowns and don't have time to test (and even if I did, wouldn't know exactly where to start looking). I may come back to this issue if it lends itself to any exploits.

**Comment by DahFrozen** [comment]
@RoboMWM it does lead to exploits. EnderPearls thrown from the offhand aren't cancelled by the plugin's cooldown or antiglitch feature. This means players can glitch, exploit and such by simply throwing Enderpearls from their offhand.

**Comment by RoboMWM** [comment]
No, I don't see how it can lead to exploits - afaik Fbasics already corrects Minecraft's erraneous teleportation calculation. I have not tested attempting to throw enderpearls through doors (I don't think the client determines projectile collision though), which is literally the only possible exploit I can think of.
